internal/app/connectconformance/results.go

```diff
@@ -471,13 +471,27 @@ func checkError(expected, actual *conformancev1.Error) multiErrors {
 	}
 
 	var errs multiErrors
-	if expected.Code != actual.Code {
-		errs = append(errs, fmt.Errorf("actual error code %d (%s) does not match expected code %d (%s)",
-			actual.Code, connect.Code(actual.Code).String(), expected.Code, connect.Code(expected.Code).String()))
+	if expected.Code != actual.Code && !inSlice(actual.Code, otherCodes) {
+		expectedCode := fmt.Sprintf("%d (%s)", expected.Code, connect.Code(expected.Code).String())
```

Comment:
Could avoid initializing `expectedCode` for the case where `len(otherCodes) > 0`

Reply:
But we actually use this initialized value in that case, at line 478 below. So it seemed simplest to me to construct it once instead of having to do it 2x from both the `if` and `else` cases.

Reply:
Thanks, didn't catch that usage. I have a feeling this could be done a bit more elegantly (maybe looping over `append(expected, otherCodes...)` and avoiding the need to use `or` in the phrasing. But it's fine.

Reply:
I reformulated, and also added a test to make sure it comes up with the expected string to present to users. Better?

Reply:
Looks great!